argon2/src/block.rs

```diff
@@ -154,3 +154,47 @@ impl Zeroize for Block {
         self.0.zeroize();
     }
 }
+
+/// Custom implementation of `Box<[Block]>` until `Box::try_new_zeroed_slice` is stabilized.
+#[cfg(feature = "alloc")]
+pub(crate) struct Blocks {
+    p: core::ptr::NonNull<Block>,
+    len: usize,
+}
+
+#[cfg(feature = "alloc")]
+impl Blocks {
+    pub fn new(len: usize) -> Option<Self> {
+        use alloc::alloc::{Layout, alloc_zeroed};
+        use core::ptr::NonNull;
+
+        if len == 0 {
+            return None;
+        }
+        
+        let layout = Layout::array::<Block>(len).ok()?;
+        // SAFETY: e use `alloc_zeroed` correctly
```

Comment:
I'm not sure what this is trying to say?

Reply:
Fixed.